backend/app/database.py

```
@@ -330,13 +349,34 @@ def seed_default_settings():
+    Returns admin id if successful.
+    
+    Raises:
+        ValueError: If an admin already exists
+    """
     with get_cursor() as cursor:
+        # Check if any admin already exists
+        cursor.execute("SELECT COUNT(*) FROM admins")
+        admin_count = cursor.fetchone()[0]
+        
+        if admin_count > 0:
+            raise ValueError("Instance already has an admin. Only one admin per instance is allowed.")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)